chia/plotting/manager.py

```diff
@@ -171,11 +171,12 @@ def __exit__(self, exc_type, exc_value, exc_traceback):
         self._lock.release()
 
     def reset(self):
```

Comment:
are you confident this is never called from the main thread?
If it is, this really needs to be an async mutex, that puts the co-routine to sleep, rather than the whole thread.
mixing regular threading and mutexes with co-routines makes me nervous about accidentally lock up threads.

Reply:
it looks like it's called here: https://github.com/xdustinface/chia-blockchain/blob/1f83bf97d71e8660632bafc7f4e2d7c2bc004a60/chia/plotting/manager.py#L319
which is a separate thread. However I think it might warrant some warning comments. I can see someone accidentally calling this from the main thread and it slipping through reviews.

Reply:
Whats the exact problem when calling it from the main thread if it eventually gets released? We lock from the main thread when populating the coroutine list for lookups here https://github.com/Chia-Network/chia-blockchain/blob/467478ef754d1bb43ceadfa5729e35029edc3f91/chia/harvester/harvester_api.py#L173

Reply:
the async mutexes don't block the thread when you acquire them, they de-schedule the co-routine and adds it to the wait-list of the mutex. Once it's its turn is it scheduled to run again. i.e. other co-routines can make progress during that time.

Reply:
Sure but this is only used by the harvester which only has the farmer connection and no blockchain to maintain...its purpose is to process signage points so isn't it reasonable to just wait for the plot manager to be ready if we anyway can't do something else, no? I agree though that all this signage point processing code needs some love but this is not related to this PR i guess.